.github/workflows/main.yml

```diff
@@ -90,18 +102,37 @@ jobs:
     - name: Use strawberry perl
       if: startsWith(matrix.os, 'windows')
       run: echo OPENSSL_SRC_PERL=C:/Strawberry/perl/bin/perl >> $GITHUB_ENV
-    - run: |
+    - name: Run tests (not Windows)
+      if: "!startsWith(matrix.os, 'windows')"
+      run: |
         set -e
         cargo generate-lockfile
         ./ci/run-docker.sh ${{ matrix.target }}
-      if: "!startsWith(matrix.os, 'windows')"
-      name: Run tests (not Windows)
-    - run: |
+    - name: Download nasm.exe (Windows)
+      if: startsWith(matrix.os, 'windows') && matrix.nasm_exe == 'installed'
+      run: |
+        WINNASMVERSION='2.15.05'
+        curl -O https://www.nasm.us/pub/nasm/releasebuilds/${WINNASMVERSION}/win64/nasm-${WINNASMVERSION}-win64.zip
+        unzip nasm-${WINNASMVERSION}-win64.zip
+        mv nasm-${WINNASMVERSION} nasm_win64
+    - name: Set NO_NASM env var (Windows)
+      if: startsWith(matrix.os, 'windows') && matrix.env_no_nasm == 'set'
+      run: echo OPENSSL_RUST_NO_NASM=1 >> $GITHUB_ENV
+    - name: Run tests (Windows)
+      if: startsWith(matrix.os, 'windows')
+      run: |
+        if [ '${{ matrix.nasm_exe }}' = 'installed' ] ; then
+        export PATH=$PWD/nasm_win64:$PATH
```

Comment:
Why do I add nasm to PATH in this step instead of in `Download nasm.exe`?
Because using `echo PATH=$PWD/xxxxxx:$PATH >> $GITHUB_ENV` in previous step will cause an error when configuring.

Reply:
For adding to `PATH` you'll want to use a different command -- https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#adding-a-system-path